src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -77,22 +82,41 @@ object DataTypes {
     @JsonScalaEnumeration(classOf[OutputTypeRef]) output: Option[OutputType]
   ) {
     /** Method which validates query fields, as jackson runs on top of runtime reflection so NPE can happen if fields are missing */
-    def validate(entity: String): Either[List[QueryValidationError], Query] = {
+    def validate(entity: String, tezosPlatformDiscovery: TezosPlatformDiscoveryOperations)(implicit ec: ExecutionContext):
+    Future[Either[List[QueryValidationError], Query]] = {
+      import cats.implicits._
+
       val query = Query().patchWith(this)
 
       val invalidQueryFields = query
         .fields
-        .filterNot(field => TezosPlatformDiscoveryOperations.areFieldsValid(entity, Set(field)))
-        .map(InvalidQueryField)
+        .map(field => tezosPlatformDiscovery.areFieldsValid(entity, Set(field)).map(_ -> field))
```

Comment:
I don't understand this code. Method ``areFieldsValid`` return ``Future[Boolean]`` but you ignore the result and map it to its original value. The same is in the method below.

Reply:
Look again at the arrow ;)

Reply:
hahaha :)